docs/source/configuration/slots.md

```diff
@@ -0,0 +1,260 @@
+---
+myst:
+  html_meta:
+    "description": "Slots are insertion points in the Volto rendering tree structure."
+    "property=og:description": "Slots are insertion points in the Volto rendering tree structure."
+    "property=og:title": "Slots"
+    "keywords": "Volto, Plone, frontend, React, configuration, slots, viewlets"
+---
+
+# Slots
+
+Slots are insertion points in the Volto rendering tree structure.
+This concept is inherited from the Plone Classic UI {doc}`plone:classic-ui/viewlets`.
+
+
+## Anatomy
+
+Slots are named, and they can contain a list of different slot components.
+Slot components are also named, and they are registered in the configuration registry using a specific API for slots.
+
+The main trait of a slot component is that its renderer is controlled by a list of conditions called {term}`predicates`.
+Multiple slot components can be registered under the same name, as long as they have different predicates.
+
+The following tree structure diagram illustrates these concepts.
+
+```text
+Slot (`toolbar`)
+├── SlotComponent (`edit`)
+│   ├── predicates (only appear in `/de/about`)
+│   ├── predicates (only appear if the content type is either a `Document` or `News Item`)
+│   └── no predicates (default when all predicates return `false`)
+├── SlotComponent (`contents`)
+└── SlotComponent (`more`)
+```
+
+At the root of the tree, there is a slot named `toolbar`.
+It contains three slot components, named `edit`, `contents`, and `more`.
+`edit` contains two predicates and a default for when all of its predicates return `false`.
```

Comment:
I don't understand the default part. How can one slot component have both 2 predicates and no predicates? Doesn't the one with no predicates need to be a separate slot component that is also named "edit"?

Reply:
Maybe if I remove the "no predicates", leaving the parent with a default, then that would help? I agree the "no predicates" appearing as a predicate in the tree is confusing.

Reply:
@stevepiercy I'm not confident I understand how it's intended to work, so I'd like @sneridagh's input here.

Reply:
For `edit` there's three slot components declared, one of them has no predicates defined. If the ones with predicates do not match, the one without predicates is returned. Not sure if call it "default". We can also modify the entry in the tree to be more precise.

Reply:
Maybe this?

```
`edit` contains three slot components, two of which have predicates and a third without predicates.
When the slots with predicates both return `false`, then third slot is returned.
```